src/groups/mqb/mqbc/mqbc_storageutil.cpp

```diff
@@ -200,20 +200,16 @@ bool StorageUtil::loadUpdatedAppInfos(AppInfos*       addedAppInfos,
     return true;
 }
 
-void StorageUtil::registerQueueDispatched(
-    mqbs::FileStore*         fs,
-    mqbs::ReplicatedStorage* storage,
-    const bsl::string&       clusterDescription,
-    int                      partitionId,
-    const AppInfos&          appIdKeyPairs)
+int StorageUtil::registerQueueDispatched(mqbs::FileStore*         fs,
+                                         mqbs::ReplicatedStorage* storage,
+                                         const AppInfos& appIdKeyPairs)
 {
     // executed by *QUEUE_DISPATCHER* thread with the specified 'partitionId'
 
     // PRECONDITIONS
     BSLS_ASSERT_SAFE(fs->inDispatcherThread());
     BSLS_ASSERT_SAFE(fs);
```

Comment:
Noticed that the order of checks is incorrect.
First the pointer is dereferenced, after this it is checked against NULL